xbmc/Application.cpp

```diff
@@ -2846,6 +2839,12 @@ bool CApplication::Cleanup()
     delete m_network;
     m_network = NULL;
 
+    if (m_ServiceManager)
```

Comment:
That seems like something that should be fixed but then it's an understandable design.